src/Domain/Entities/Order.cs

```diff
@@ -1,15 +1,27 @@
 ﻿using Domain.Common;
 using System;
 using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
 
 namespace Domain.Entities
 {
     public class Order : BaseEntity
     {
+        [Required]
         public int UserId { get; set; }
+
+        [Required]
         public string ShippingAddress { get; set; }
+
+        [Required]
         public string Phone { get; set; }
+
+        [Required]
+        [RegularExpression(@"^\d+\.\d{0,2}$")]
+        [Range(0, 9999999999999999.99, ErrorMessage = "Invalid Delivery Charge; Max 18 digits")]
```

Comment:
Both RegExp and Range is not necessary. Use RegExp `^\d{1,18}(\.\d{1,2}){0,1}$` or the Range.